doc/development/deprecations.rst

```diff
@@ -6,6 +6,18 @@ Deprecations
 Pending deprecations
 --------------------
 
+* The ``qml.specs`` dictionary will no longer support direct key access to certain keys. Instead
```

Comment:
What will happen when `qml.specs` is used? The user gets all the warnings?

Reply:
Only if they call specs for a specific attribute that is duplicated. So `qml.specs['num_trainable_params']` will not raise any warning but `qml.specs["depth"]` will raise a deprecation warning.

Reply:
How about `qml.specs` without any attribute specification? e.g.,

`>>> qml.specs(circuit)(params)`

There will be no warning I guess? Maybe we should add a general warning?